server/fleet/host_certificates.go

```diff
@@ -241,11 +242,23 @@ func ExtractDetailsFromOsqueryDistinguishedName(str string) (*HostCertificateNam
 		case "O":
 			details.Organization = strings.Trim(value, " ")
 		case "OU":
-			details.OrganizationalUnit = strings.Trim(value, " ")
+			// osquery is inconsistent in how it reports certs with multiple OUs; sometimes it
+			// concatenates them all joined by `+OU=` separator within the same `/` delimited
+			// string, other times it provides multiple `/` delimited strings that each contain
+			// distinct OU values. For example, compare the following two lines:
+			//   /OU=SomeValue/OU=fleet-a3d5d6f4c-819e-4159-9a42-0d6243a80ff8/CN=SomeName
+			//   /OU=SomeValue+OU=fleet-a0c039413-d0c7-4b1f-9488-b93c865351ac/CN=SomeName
+			//
+			// To handle both cases, we collect all OU values and join them with `+OU=` below.
+			// We should probably reconsider our approaches for normalization of cert data
+			// across the board.
+			// FIXME: How should this work with the edge case covered by PR 33152 (see line 224 above)?
+			ouParts = append(ouParts, strings.Trim(value, " "))
```

Comment:
This addresses potential "last-one-wins" issues where earlier values could be dropped.